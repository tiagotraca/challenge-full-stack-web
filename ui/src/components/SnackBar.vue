<template>
  <v-snackbar
    :color="color"
    outlined
    :value="true"
    v-model="snackbar"
    :timeout="timeout"
  >
    {{ text }}

    <template v-slot:action="{ attrs }">
      <v-btn color="blue" text v-bind="attrs" @click="snackbar = false">
        Close
      </v-btn>
    </template>
  </v-snackbar>
</template>

<script>
export default {
  data: () => ({
    snackbar: false,
    text: "My timeout is set to 2000.",
    timeout: 3000,
    color: "success",
  }),
  watch: {
    "$store.state.snackbar": function () {
      if (this.$store.state.snackbar) {
        this.snackbar = this.$store.state.snackbar;
        this.text = this.$store.state.message;
        this.color = this.$store.state.color;
      }
      this.$store.dispatch("modifyState", false);
    },
  },
};
</script>
