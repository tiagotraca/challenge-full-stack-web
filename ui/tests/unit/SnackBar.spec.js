import { createLocalVue, mount } from "@vue/test-utils";
import SnackBar from "@/components/SnackBar.vue";
import Vuetify from "vuetify";

const localVue = createLocalVue();
describe("SnackBar.vue", () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  it("should have a custom title and match snapshot", () => {
    const wrapper = mount(SnackBar, {
      localVue,
      vuetify,
    });
    const title = wrapper.getComponent(".v-snack__content");
    expect(title.text()).toBe("My timeout is set to 2000.");

    expect(wrapper.html()).toMatchSnapshot();
  });
});
