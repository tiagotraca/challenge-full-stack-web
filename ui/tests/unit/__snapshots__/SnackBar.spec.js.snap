// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackBar.vue should have a custom title and match snapshot 1`] = `
<div class="v-snack v-snack--bottom">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light success--text" style="display: none;">
      <div role="status" aria-live="polite" class="v-snack__content">
        My timeout is set to 2000.

      </div>
      <div class="v-snack__action "><button type="button" class="v-btn v-btn--text theme--light v-size--default blue--text v-snack__btn"><span class="v-btn__content">
      Close
    </span></button></div>
    </div>
  </transition-stub>
</div>
`;
